torchtune/datasets/multimodal/_the_cauldron.py

```diff
@@ -91,7 +91,10 @@ def __call__(self, sample: Mapping[str, Any]) -> Mapping[str, Any]:
                 Message(
                     role="user",
                     content=[
-                        {"type": "image"},
+                        {
+                            "type": "image",
+                            "content": sample[self._column_map["images"]],
```

Comment:
I mean it can - but it should be consistent

Reply:
Damn, if this is coming in as a PIL image, then it might make sense to go ahead with the changes to load image into Message as a PIL image

Reply:
ha... thinking about it a bit more, I suppose there's no actual memory hit. if you're using multimodal datasets you will certainly need to load the image at some point. then again, changing my mind after five minutes hurts my pride just a little

Reply:
I don't really want a message to be ```Union[Path, str, PIL.Image]```

That seems like a recipe for confusion.

Reply:
ALRIGHT F IT, here's the plan:

I was a big ol dumb dumb. This stuff is gunna go. Messages will now hold PIL images. This will be loosely typed in Messages so that we don't need to actually import PIL and Messages can stay free of that dependency hell. Then, the onus for providing a proper PIL image to the Message class is on the dataset builder and the generation recipe. For generation, we could just have a separate one for multimodal or require PIL for everything. For dataset builder, everything is in the multimodal folder., which protects us from imports (thank for the forward thinking @RdoubleA)

This makes the most logical sense IMO. 

data entry points (datasets, generation) load in the PIL image -> feed to message in proper format -> message is used by model transform however it wants. 

Okay? Okay.

Reply:
Idk what just happened here but I'm on board

Reply:
> ALRIGHT F IT

you right, I'm on board